test/infrastructure/docker/capd-up.sh

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Helper script to tilt up an environment for working with CAPD.
+
+# Make sure we run from the root of the repo
+cd "$(git rev-parse --show-toplevel)" || ( \
+    echo "!! Unable to find repo root, make sure to run from within cluster-api repo."
+    exit 1)
+
+# Verify we have all the expected prerequisites
+echo "++ Checking prerequisites..."
+PREREQS=( docker kind kustomize tilt envsubst kubectl )
```

Comment:
Not sure if it's worth it but for kind/kustomize/kubectl you could reuse the scripts from: https://github.com/kubernetes-sigs/cluster-api/tree/master/hack 

They also validate the version. This could be useful, e.g. for non-obvious errors with old kustomize versions

Reply:
Thansk @sbueringer, I hadn't looked in to those scripts yet. They do have some useful checks. One issue I see though is that they check for the binary to be under hack/tools/bin. I think (and correct me if I'm wrong) tilt looks for these executables on $PATH. So I think even if they were built for these scripts to pass, we might still fail doing a `tilt up` like what is done here.

Maybe worth seeing if there is something more we can do with a future update?

Reply:
@sbueringer point make sense, this will avoid some errors hard to debug.
Also, AFAIK tilt uses files under hack/tools/bin at least for Kustomise and envsubst, which already created such problems in the past.

Reply:
Oh, that's great. Then if we decide this script is something we want in-repo, I'll update to call out to these scripts.